pkgs/servers/search/elasticsearch/plugins.nix

```diff
@@ -36,6 +36,21 @@ in {
     };
   };
 
+  elasticsearch_cloud_aws = esPlugin rec {
+    name = "elasticsearch-cloud-aws-${version}";
+    pluginName = "elasticsearch-cloud-aws";
+    version = "2.4.2";
+    src = fetchurl {
+      url = "https://github.com/elastic/elasticsearch-cloud-aws/archive/v2.4.2.zip";
```

Comment:
What about this url
https://github.com/elastic/elasticsearch-cloud-aws/archive/0792440f6ff1c1a2d0002dd4497dd5a3123f867d.zip

On Sun, Jun 7, 2015 at 10:27 PM Benjamin Staffin <notifications@github.com>
wrote:

> In pkgs/servers/search/elasticsearch/plugins.nix
> <https://github.com/NixOS/nixpkgs/pull/8181#discussion_r31880261>:
>
> > @@ -36,6 +36,21 @@ in {
> >      };
> >    };
> >
> > +  elasticsearch_cloud_aws = esPlugin rec {
> > +    name = "elasticsearch-cloud-aws-${version}";
> > +    pluginName = "elasticsearch-cloud-aws";
> > +    version = "2.4.2";
> > +    src = fetchurl {
> > +      url = "https://github.com/elastic/elasticsearch-cloud-aws/archive/v2.4.2.zip";
>
> The trouble with that is the elasticsearch plugin manager expects to be
> passed a zip file, and fetchFromGitHub always unpacks the archive after
> downloading it. You could do something like this, though it's a bit hacky.
>
>   elasticsearch_cloud_aws = esPlugin rec {
>     name = "elasticsearch-cloud-aws-${version}";
>     pluginName = "elasticsearch-cloud-aws";
>     version = "2.4.2";
>     src = fetchFromGitHub {
>       owner = "elastic";
>       repo = "elasticsearch-cloud-aws";
>       rev = "0792440f6ff1c1a2d0002dd4497dd5a3123f867d";
>       sha256 = "etc";
>     };
>     # fetchFromGitHub unpacks, but we want a zip. Can't reliably use fetchurl because
>     # github sometimes regenrates zips and you'll get inconsistent checksums :(
>     preConfigure = ''      newsrc=$PWD/v2.4.2.zip      (cd $src/../ && ${pkgs.zip}/bin/zip -r $newsrc $(basename $src))      src=$newsrc    '';
>   };
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/8181/files#r31880261>.
>